### PR TITLE
Let query.Sort/query.Window apply to clear queries

### LINKED ISSUE #25: Clear ignores Window
While working on rs/rest-layer#127 I discovered that Clear ignores Window (limit / offset). I suppose it should handle it equally to Find?

https://github.com/rs/rest-layer-mongo/blob/128c54c62a7d09baba781c232724759a5f1057fc/mongo.go#L199

### PR DESCRIPTION
**Pull request by smyrman**
Fixes #25.

TODO:
- [x] Ensure passing unit tests for skip / limit
- [x] Manually test together with rs/rest-layer#186

### CODE REVIEW

File: mongo_test.go [hunk 1]
```diff
@@ -229,13 +242,82 @@ func TestClear(t *testing.T) {
...
 
 	q, err = query.New("", `{id:"2"}`, "", nil)
 	if assert.NoError(t, err) {
 		deleted, err := h.Clear(context.Background(), q)
 		assert.NoError(t, err)
 		assert.Equal(t, 1, deleted)
 	}
+	assertCollectionIDs(t, s.DB(dbName).C(dbName), []string{"1"})
+}
+func TestClearLimit(t *testing.T) {
```
Comment (smyrman) [inline mongo_test.go]: The update of the tests is done in a lazy way atm. Ideally we would rewrite this file to drop the testify dependency as well as use table tests for less code/check duplication. However, I did not want to spend any time on that for this PR.

**Comment by smyrman** [comment]
Tested to  work, ready for review.

File: mongo.go [hunk 2]
```diff
@@ -195,30 +195,55 @@ func (m Handler) Delete(ctx context.Context, item *resource.Item) error {
...
 	}
 	defer m.close(c)
+
+	if q.Window != nil {
+		// RemoveAll does not allow skip and limit to be set. To workaround this
+		// we do an additional pre-query to retrieve a sorted and sliced list of
+		// unique IDs that we will remove in the next step.
+		//
+		// PS! This solution is limted to handle the case where the BSON
+		// encoding of the query is less than 16MiB.
```
[the remark this answers is not in the document]
Comment (smyrman) [reply mongo.go]: A qury may not be larger than the BSON max doc size. I can rephrase and leave a link https://docs.mongodb.com/manual/reference/limits/#bson-documents

[the remark this answers is not in the document]
Comment (rs) [reply mongo.go]: Would be nice.

**Comment by smyrman** [comment]
@rs, ready for review again.

The amended commit updates only the comment inside the Clear method, as well as the doc-string for the method.

**Pull request merged**